### ISSUE TITLE
HTTPS filtering fails on https://outlook.live.com/owa/

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/CoreLibs#597]

**Comment by sfionov** [comment]
Not reproducable anymore.

**Comment by BooBerry** [comment]
Still reproducible for me but that tends to just happen. For some reason at random filtering just fails on Outlook while it works on all other sites.

Might try a clean reinstall of both Chrome and AdGuard for Mac nightly and see if it works then.

**Comment by sfionov** [comment]
@BooBerry Thanks, will recheck it.

**Comment by BooBerry** [comment]
An update, did a clean cache of Safari and it's not working there anymore either - no Assistant or ad blocking. Will do the same to Firefox.

**Comment by BooBerry** [comment]
No go on either one. Still fails.

**Comment by BooBerry** [comment]
This keeps getting stranger. Sometimes it fails in Chrome, but once in awhile after any amount of random reloads, filtering works again... until I reload the page (or clear site data). I don't think the browser matters either, nor Outlook per-say. It seems filtering randomly fails in general in AG for Mac nightly.

**Comment by BooBerry** [comment]
I might've figured this one out finally. You're right Andrey, it's probably the service worker. Right now I'm testing this filter and it seems to be working 100% of the time...

`||outlook.live.com/mail/sw.js^$domain=outlook.live.com,important`

You guys might want to test this one for a little bit, if it doesn't break anything (as far as I can tell, it doesn't) you might consider adding it to the base filter.

**Comment by BooBerry** [comment]
Another odd way to reproduce it 100% of the time with the latest AG for Mac nightly + Chrome...

With Chrome open (and no Outlook tab open) check for filter updates in AG for Mac and after the filters are updated (I used the AG Base + Spyware + Annoyances filters in this test, and they were all updated) then open Outlook in a tab. First load there's no ad blocking (e.g. the right ad placeholder appears) and no Assistant. If you refresh the Outlook tab, it works normally.

This is the Outlook URL I use via bookmark (in case that matters): https://outlook.live.com/mail/#/inbox

This seems to only work when there's filter updates. If I check for filter updates and there aren't any, I can't reproduce the issue. That's kinda why I was wondering it was related to the filter issues with AG for Android as it's just plain odd that updating filters causes it to happen.

**Comment by ameshkov** [comment]
@BooBerry could you please reproduce the issue and then check Chrome dev tools:
https://uploads.adguard.com/up04_igurd_Example_Domain.png

Could it be that the service worker somehow slipped through while AG was reconfiguring protection?

**Comment by BooBerry** [comment]
Yeah, I've been watching that when reproducing it, and no, the service worker doesn't load at all according to the dev tools.

**Comment by ameshkov** [comment]
@sfionov @szolin guys, check this out:

1. Outlook caches the page content using a service worker (which is absolutely okay).
2. Alongside the page, it caches links to `local.adguard.org/*/contentScript.js`
3. After AG restart, that link becomes invalid due to the expired auth token.

We should figure what to do with it.

**Comment by sxgunchenko** [comment]
This may be helpful https://jakearchibald.com/2014/offline-cookbook/#network-only

**Comment by ameshkov** [comment]
Guys, again, the first thing we need to address is that cached links to content-script do not work due to expired auth token.

**Comment by szolin** [comment]
/core/pull-requests/1008

**Comment by Steve1209** [comment]
Just to let the Dev's know, a version of this problem is still happening in the nightly released tonight 6-3-2019.  See here https://forum.adguard.com/index.php?threads/adguard-assistant-disappears-and-adguard-stops-filtering.31136/
@BooBerry's  user filter suggestion seems to fix for me on Gmail

**Comment by ameshkov** [comment]
Here's an update.

There's no good generic solution to this that we can implement on the network level. 

Options are:
1. Suppressing service workers (not a very good solution)
2. Handle this with a browser extension. Once we have a native browser assistant extension, we'll be able to handle this properly.

**Comment by BooBerry** [comment]
First option could be a stealth mode option, I suppose (for those who want to block them in general)? But yeah, not a good solution at all.

The second option is probably the way to go, IMO.

**Comment by sfionov** [comment]
Merged, will be delivered in 1.23.0 release.
